### PR TITLE
Replace "Initialize tools" to only download docker files

### PR DESCRIPTION
**Pull request by smile21prc**
This way "Initialize tools" will be done later INSIDE docker, instead of
OUTSIDE. This fixes issue:

https://github.com/dotnet/core-eng/issues/1544

The inline script is:

if command -v curl > /dev/null; then
    curl --retry 10 -sSL -o cleanup-docker.sh $(PB_VsoDockerFileDirUrl)cleanup-docker.sh
    curl --retry 10 -sSL -o init-docker.sh $(PB_VsoDockerFileDirUrl)init-docker.sh
else
    wget $(PB_VsoDockerFileDirUrl)cleanup-docker.sh -q -O cleanup-docker.sh
    wget $(PB_VsoDockerFileDirUrl)init-docker.sh -q -O init-docker.sh
fi

### CODE REVIEW

File: buildpipeline/DotNet-CoreFx-Trusted-Linux.json [hunk 1]
```diff
@@ -71,21 +71,26 @@
...
-        "filename": "$(PB_GitDirectory)/init-tools.sh",
-        "arguments": "",
-        "workingFolder": "$(PB_GitDirectory)",
-        "failOnStandardError": "false"
+        "type": "InlineScript",
+        "scriptPath": "",
+        "args": "",
+        "cwd": "",
+        "failOnStandardError": "false",
+        "script": "if command -v curl > /dev/null; then\n    curl --retry 10 -sSL -o cleanup-docker.sh $(PB_VsoDockerFileDirUrl)cleanup-docker.sh\n    curl --retry 10 -sSL -o init-docker.sh $(PB_VsoDockerFileDirUrl)init-docker.sh\nelse\n    wget $(PB_VsoDockerFileDirUrl)cleanup-docker.sh -q -O cleanup-docker.sh\n    wget $(PB_VsoDockerFileDirUrl)init-docker.sh -q -O init-docker.sh\nfi"
```
Comment (weshaggard) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux.json]: Lets put these in some directory that will be ignored by git perhaps "Tools/dockerscripts" or something like that.

File: buildpipeline/DotNet-CoreFx-Trusted-Linux.json [hunk 2]
```diff
@@ -402,6 +407,9 @@
     "PB_VsoRepositoryName": {
       "value": "DotNet-CoreFX-Trusted"
     },
+    "PB_VsoDockerFileDirUrl": {
+      "value": "https://$(PB_VsoAccountName):$(VsoPassword)@devdiv.visualstudio.com/DevDiv/_git/DotNet-Buildtools-Trusted/src/Microsoft.DotNet.Build.Tasks/PackageFiles/scripts/docker/"
```
Comment (weshaggard) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux.json]: While I'm OK with this as temp fix we need a more long term fix that doesn't depend on a unversioned repo endpoint. Can you please file a core-eng issue on this?

File: buildpipeline/DotNet-CoreFx-Trusted-Linux.json (hunk 1 above)
Comment (smile21prc) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux.json]: Great suggestion. Added var:
    "DockerFileDir": {
      "value": "tools/dockerscripts"
    },

And changed to download those files to that folder instead:
if [[ ! -d $(DockerFileDir) ]]; then
    mkdir -p $(DockerFileDir)
fi

if command -v curl > /dev/null; then
    curl --retry 10 -sSL -o $(DockerFileDir)/cleanup-docker.sh $(VsoDockerFileDirUrl)cleanup-docker.sh
    curl --retry 10 -sSL -o $(DockerFileDir)/init-docker.sh $(VsoDockerFileDirUrl)init-docker.sh
else
    wget $(VsoDockerFileDirUrl)cleanup-docker.sh -q -O $(DockerFileDir)/cleanup-docker.sh
    wget $(VsoDockerFileDirUrl)init-docker.sh -q -O $(DockerFileDir)/init-docker.sh
fi

The calling site in "Initialize Docker" is updated as well.

File: buildpipeline/DotNet-CoreFx-Trusted-Linux.json (hunk 2 above)
Comment (smile21prc) [inline buildpipeline/DotNet-CoreFx-Trusted-Linux.json]: Sure @weshaggard . I've created https://github.com/dotnet/core-eng/issues/1554 on it.

Could you please take a second look on the updated version?

**Comment by smile21prc** [comment]
test Linux x64 Release Build

**Comment by smile21prc** [comment]
Test Tizen armel Debug Build

**Comment by smile21prc** [comment]
test Windows x64 Debug Build

**Comment by smile21prc** [comment]
Thanks @weshaggard , the issue to track long term fix is https://github.com/dotnet/core-eng/issues/1554.

**Pull request merged**